src/test/codegen/unwind-extern.rs

```diff
@@ -0,0 +1,15 @@
+// compile-flags: -C opt-level=0
+
+#![crate_type = "lib"]
+#![feature(unwind_attributes)]
+
+// make sure these all do *not* get the attribute
+// CHECK-NOT: nounwind
+
+pub extern fn foo() {} // right now we don't abort-on-panic, so we also shouldn't have `nounwind`
```

Comment:
This is the key test, making sure that when we don't add the abort-on-panic shim, we also do not add `nounwind`.

Reply:
I think this test is incorrect on nightly, where this function should always be `nounwind`, independently of the optimization level, but correct for stable where rustc should not add the attribute.

Reply:
Why should this be `nounwind` on nightly...? We don't insert an abort-on-panic shim any more.

Reply:
> We don't insert an abort-on-panic shim any more.

Why not? This is an `extern "C" fn` definition, and on nightly, a panic that tries to unwind it should abort. (On stable Rust, a panic should be able to unwind this function, and therefore it cannot be `nounwind` there).

Reply:
See above -- and on top of that, it seems silly to break people's code on nightly only. I don't think there is precedence for making nightly vs stable Rust a different language. Stable Rust is just a snapshot from the time of release.

Reply:
> Why not?

This landed recently in https://github.com/rust-lang/rust/issues/62825. There's been tons of discussion. All of that is out of scope of this PR.

Reply:
I did not know that this changed on nightly. I thought the nightly behavior was still the "old" one, where it differed from stable (this has been the state for a long while).

Reply:
I referenced the PR that changed that in the description here, but maybe I should have been clearer.

Marking this discussion as resolved then.